src/schemes/fluid/shifting_techniques.jl

```diff
@@ -58,44 +70,360 @@ cost and improve the stability of the scheme.
+
+        new{integrate_shifting_velocity,
+            update_everystage,
+            modify_continuity_equation,
+            second_continuity_equation_term,
+            modify_momentum_equation,
+            compute_v_max, typeof(v_factor)}(v_factor)
+    end
+end
+
+"""
+    ParticleShiftingTechniqueSun2017(; kwargs...)
+
+Particle Shifting Technique by [Sun et al. (2017)](@cite Sun2017).
+Following the original paper, the callback is applied in every time step and not
+in every stage of a multi-stage time integration method to reduce the computational cost.
+
+This is a convenience constructor for:
+```jldoctest; output = false
+ParticleShiftingTechnique(integrate_shifting_velocity=false,
+                          update_everystage=false,
+                          modify_continuity_equation=false,
+                          second_continuity_equation_term=false,
+                          modify_momentum_equation=false,
+                          v_max_factor=1, sound_speed_factor=0)
+
+# output
+ParticleShiftingTechnique{false, false, false, false, false, true, Int64}(1)
+```
+
+See [ParticleShiftingTechnique](@ref ParticleShiftingTechnique) for all available options.
 
-# Zero because PST is applied in a callback
-@inline function delta_v(system, ::ParticleShiftingTechnique, particle)
+# Keywords
+- `kwargs...`: All keywords are passed to the main constructor.
+
+# Examples
+```jldoctest; output = false
+shifting_technique = ParticleShiftingTechniqueSun2017()
+
+# output
+ParticleShiftingTechnique{false, false, false, false, false, true, Int64}(1)
+```
+
+!!! warning
+    The Particle Shifting Technique needs to be disabled close to the free surface
+    and therefore requires a free surface detection method. This is not yet implemented.
+    **This technique cannot be used in a free surface simulation.**
+"""
+function ParticleShiftingTechniqueSun2017(; kwargs...)
+    return ParticleShiftingTechnique(; integrate_shifting_velocity=false,
+                                     update_everystage=false,
+                                     modify_continuity_equation=false,
+                                     second_continuity_equation_term=false,
+                                     modify_momentum_equation=false,
+                                     v_max_factor=1, sound_speed_factor=0,
+                                     kwargs...)
+end
+
+"""
+    ConsistentShiftingSun2019(; sound_speed_factor=0.1, kwargs...)
+
+Consistent Particle Shifting Technique by [Sun et al. (2019)](@cite Sun2019).
+
+This is a convenience constructor for:
+```jldoctest; output = false
+ParticleShiftingTechnique(integrate_shifting_velocity=true,
+                          update_everystage=true,
+                          modify_continuity_equation=true,
+                          second_continuity_equation_term=true,
+                          modify_momentum_equation=true,
+                          v_max_factor=0, sound_speed_factor=0.1)
```

Comment:
@sloede Is this a very elegant solution or rather ugly?
There is one constructor `ParticleShiftingTechnique` that takes a lot of kwargs for maximum flexibility. Then there are these convenience constructors to conveniently get the configurations of several papers.